### PR TITLE
incorporation localization of texts using of the library i18n

### LINKED ISSUE #91: Add i18n / localizations
## Overview
Teloscan presently has all strings in the UI hardcoded. It would be great if the application instead supported i18n so that site translations could be possible

## Acceptance criteria
- a list of all magic strings in the application
    - including html `label`, `alt`, `placeholder`, `hint`, `aria-label`, etc. attributes

## Tasks

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #91

## Description
Telos scan needs to include internationalization and for that i18n was incorporated and all hard-coded texts were changed for a call to the $t() function with the corresponding path.

## Requirements 
This PR needs to be merged after merging the [incorporation of the library i18n PR](https://github.com/telosnetwork/teloscan/pull/292) 

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages

### CODE REVIEW

**Comment by donnyquixotic** [comment]
@Viterbo check syntax in templates, text fields aren't rendering:
image

**Review by donnyquixotic — CHANGES_REQUESTED** [review]
See comment

File: src/components/Transaction/LogsViewer.vue [hunk 1]
```diff
@@ -14,10 +14,12 @@
...
+            {{ $t('components.transaction.human_readable') }}
             <small v-if="!allVerified">
                 <q-icon name="info" class="q-mb-xs q-ml-xs" size="14px"/>
-                <q-tooltip>Verify the related contract for each log to see its human readable version</q-tooltip>
+                <q-tooltip>
+                    {{ $t('components.transaction.verify_related_contract') }}
+                </q-tooltip>
             </small>
         </div>
         <div class="col-12">
```
Comment (karynemayer) [inline src/components/Transaction/LogsViewer.vue:25]: Line 59

File: src/components/TransactionTable.vue [hunk 2]
```diff
@@ -69,6 +31,44 @@ export default {
...
+            {
+                name: 'value',
+                label: this.$t('components.value_transfer'),
+                align: 'left',
+            },
+        ];
+
         return {
             rows: [],
             columns,
```
Comment (karynemayer) [inline src/components/TransactionTable.vue:74]: line 144
line 191
line 195

File: src/components/TransferTable.vue [hunk 3]
```diff
@@ -70,6 +37,40 @@ export default {
...
+                name: 'token',
+                label: this.$t('components.token'),
+                align: 'left',
+            },
+        ];
+
+
         return {
             rows: [],
             columns,
```
Comment (karynemayer) [inline src/components/TransferTable.vue:76]: Line 105

File: src/components/TransferTable.vue [hunk 4]
```diff
@@ -156,7 +157,7 @@ export default {
                             if (token && typeof token.decimals === 'number') {
                                 valueDisplay = formatWei(log.data, token.decimals)
                             } else {
-                                valueDisplay = 'Unknown precision';
+                                valueDisplay = this.$t('components.unknown_precision');
                             }
                         } else {
                             let tokenId = (this.tokenType === 'erc1155') ? BigNumber.from(log.data.substr(0, TOKEN_ID_TRUNCATE_LENGTH)).toString() : BigNumber.from(log.topics[3]).toString();
```
Comment (karynemayer) [inline src/components/TransferTable.vue:163]: line 167
line 181

File: src/components/inputs/AddressInput.vue [hunk 5]
```diff
@@ -42,9 +42,9 @@ export default {
...
         rules() {
-            const errMessageInvalidLength = 'An address must be exactly 40 characters, not including "0x"';
-            const errMessageStartsWith0x = 'An address must begin with 0x';
-            const errMessageInvalidInput = 'Entry contains invalid characters';
+            const errMessageInvalidLength = this.$t('components.inputs.invalid_address_length');
+            const errMessageStartsWith0x = this.$t('components.inputs.invalid_address_start');
+            const errMessageInvalidInput = this.$t('components.inputs.invalid_address_characters');
 
             const startsWith0xRegex = /(^0x)|(^$)/;
             const addressRegex = /(^0x[0-9a-fA-F]{40}$)|(^$)/;
```
Comment (karynemayer) [inline src/components/inputs/AddressInput.vue:50]: line 59

File: src/pages/AccountAddress.vue [hunk 6]
```diff
@@ -126,9 +126,9 @@ export default {
...
                 this.title = this.contract.getName();
             } else if (this.isContract) {
-                this.title = 'Contract';
+                this.title = this.$t('pages.contract');
             } else {
-                this.title = 'Account';
+                this.title = this.$t('pages.account');
             }
 
             this.accountLoading = false;
```
Comment (karynemayer) [inline src/pages/AccountAddress.vue:134]: line 145

File: src/pages/Block.vue [hunk 7]
```diff
@@ -39,10 +39,10 @@ export default {
...
           .dataCardItem
-            .dataCardTile Gas used
+            .dataCardTile {{ $t('pages.gas_used') }}
             .dataCardData {{ parseInt(blockData.gasUsed, 16) }}
           .dataCardItem
-            .dataCardTile Transactions
+            .dataCardTile {{ $t('pages.transactions') }}
             .dataCardData {{ blockData.transactions.length || 0 }}
           .dataCardItem
             .dataCardTile
```
Comment (karynemayer) [inline src/pages/Block.vue:48]: could you delete line 50?

File: src/pages/ContractVerification.vue [hunk 8]
```diff
@@ -161,54 +161,54 @@ export default {
                 q-input(
```
Comment (karynemayer) [inline src/pages/ContractVerification.vue:161]: line 152

File: src/pages/Transaction.vue [hunk 9]
```diff
@@ -131,7 +131,7 @@ export default {
                                     console.error(`Could not retreive metadata for ${contract.address}: ${e.message}`);
```
Comment (karynemayer) [inline src/pages/Transaction.vue:131]: probably should be sent to the UI and translated as well

File: src/pages/Transaction.vue [hunk 10]
```diff
@@ -146,15 +146,15 @@ export default {
...
```
Comment (karynemayer) [inline src/pages/Transaction.vue:146]: same

File: src/pages/Transaction.vue [hunk 11]
```diff
@@ -241,7 +241,7 @@ export default {
         | Not found: {{ hash }}
```
Comment (karynemayer) [inline src/pages/Transaction.vue:241]: line 239

File: src/pages/Transaction.vue [hunk 12]
```diff
@@ -328,7 +328,7 @@ export default {
               div(class="col-3")
```
Comment (karynemayer) [inline src/pages/Transaction.vue:328]: lines 291, 298, 304, 308, 312, 316, 319, 323

File: src/pages/Transaction.vue [hunk 13]
```diff
@@ -328,7 +328,7 @@ export default {
...
                 strong {{ `From: ` }}
```
Comment (karynemayer) [inline src/pages/Transaction.vue:329]: this

File: src/pages/Transaction.vue [hunk 14]
```diff
@@ -364,9 +364,9 @@ export default {
                     span {{ formatWei(trx.value, 18) }} TLOS
```
Comment (karynemayer) [inline src/pages/Transaction.vue:364]: lines 335, 341, 347, 353, 364 and 365

File: src/pages/Transaction.vue [hunk 15]
```diff
@@ -364,9 +364,9 @@ export default {
...
             br
-            ERCTransferList( v-if="erc20Transfers.length > 0" type="ERC20" :trxFrom="trx.from" :contract="contract" :transfers="erc20Transfers")
-            ERCTransferList( v-if="erc721Transfers.length > 0" type="ERC721" :trxFrom="trx.from" :contract="contract" :transfers="erc721Transfers")
-            ERCTransferList( v-if="erc1155Transfers.length > 0" type="ERC1155" :trxFrom="trx.from" :contract="contract" :transfers="erc1155Transfers")
+            ERCTransferList( v-if="erc20_transfers.length > 0" type="ERC20" :trxFrom="trx.from" :contract="contract" :transfers="erc20_transfers")
+            ERCTransferList( v-if="erc721_transfers.length > 0" type="ERC721" :trxFrom="trx.from" :contract="contract" :transfers="erc721_transfers")
+            ERCTransferList( v-if="erc1155_transfers.length > 0" type="ERC1155" :trxFrom="trx.from" :contract="contract" :transfers="erc1155_transfers")
             div(class="fit row wrap justify-start items-start content-start")
               div(class="col-3")
                 strong {{ `Gas Price Charged: ` }}
```
Comment (karynemayer) [inline src/pages/Transaction.vue:372]: lines 372, 377, 378, 383, 388, 393, 398, 403

File: src/pages/staking/StakeForm.vue [hunk 16]
```diff
@@ -13,13 +13,13 @@
                 <q-btn
```
Comment (karynemayer) [inline src/pages/staking/StakeForm.vue:13]: line 11

File: src/pages/staking/StakeForm.vue [hunk 17]
```diff
@@ -57,38 +57,38 @@
         <q-card>
```
Comment (karynemayer) [inline src/pages/staking/StakeForm.vue:57]: line 53

File: src/pages/staking/StakeForm.vue [hunk 18]
```diff
@@ -242,16 +244,16 @@ export default {
...
                 else
-                    return 'Stake TLOS';
+                    return this.$t('pages.staking.stake_tlos');
             }
 
-            return 'Connect Wallet';
+            return this.$t('pages.staking.connect_wallet');
         },
         showClaimBanner() {
             return this.hasUnlockedTlos && !this.userDismissedBanner;
```
Comment (karynemayer) [inline src/pages/staking/StakeForm.vue:259]: line 275, 304, 378

File: src/pages/staking/Staking.vue [hunk 19]
```diff
@@ -36,21 +36,21 @@
...
                     <q-route-tab
                         name="withdraw"
                         :to="{ hash: '#withdraw'}"
                         exact
                         push
-                        label="Withdraw"
+                        :label="$t('pages.staking.withdraw')"
                         :alert="showWithdrawNotification ? 'green' : false"
                     />
                 </q-tabs>
```
Comment (karynemayer) [inline src/pages/staking/Staking.vue:56]: Please, evaluate all the `console.error` messages below this point

File: src/pages/staking/StakingStats.vue [hunk 20]
```diff
@@ -138,25 +134,17 @@ export default {
...
+                    tooltip: this.$t('pages.staking.tooltip_4', { unlockPeriod: this.unlockPeriodPretty }),
                 },
             };
         },
         unlockPeriodPretty() {
-            return formatUnstakePeriod(this.unstakePeriodSeconds);
+            return formatUnstakePeriod(this.unstakePeriodSeconds, this.$t);
         },
     },
     async created() {
```
Comment (karynemayer) [inline src/pages/staking/StakingStats.vue:150]: `console.error` message

File: src/pages/staking/UnstakeForm.vue [hunk 21]
```diff
@@ -32,16 +32,16 @@
         <q-card>
```
Comment (karynemayer) [inline src/pages/staking/UnstakeForm.vue:32]: line 28

File: src/pages/staking/UnstakeForm.vue [hunk 22]
```diff
@@ -32,16 +32,16 @@
...
-                    Heads up, you may unstake <span class="text-primary">{{ remainingDeposits }}</span> more times before
-                    you reach the maximum concurrent unstake actions. When you reach the maximum, you will need to claim
-                    unlocked TLOS to continue unstaking. If you don't have any claimable TLOS at that time, you must
-                    wait until the lock duration has elapsed before you can claim unlocked TLOS and unstake more sTLOS.
+                    {{ $t('pages.staking.confirm_unstake_2a') }}
+                    <span class="text-primary">{{ remainingDeposits }}</span>
+                    {{ $t('pages.staking.confirm_unstake_2b') }}
+
                 </p>
                 Would you like to proceed?
```
Comment (karynemayer) [inline src/pages/staking/UnstakeForm.vue:46]: this

File: src/pages/staking/UnstakeForm.vue [hunk 23]
```diff
@@ -182,7 +180,7 @@ export default {
         },
```
Comment (karynemayer) [inline src/pages/staking/UnstakeForm.vue:180]: line 179

File: src/pages/staking/UnstakeForm.vue [hunk 24]
```diff
@@ -200,9 +198,8 @@ export default {
...
         ctaText() {
             if (this.ctaIsLoading)
-                return 'Loading...';
-
-            return this.isLoggedIn ? 'Unstake sTLOS' : 'Connect Wallet';
+                return this.$t('pages.staking.loading'); // 'Loading...'
+            return this.isLoggedIn ? this.$t('pages.staking.unstake_stlos') : this.$t('pages.staking.connect_wallet');
         },
         remainingDeposits() {
             return (this.maxDeposits ?? 0) - this.deposits.length;
```
Comment (karynemayer) [inline src/pages/staking/UnstakeForm.vue:205]: `console.error` messages

File: src/pages/staking/WithdrawPage.vue [hunk 25]
```diff
@@ -66,12 +66,12 @@
...
+                {{ $t('pages.staking.withdraw_tlos') }}
             </q-btn>
         </div>
     </div>
     <div v-if="resultHash" class="transaction-notification col-sm-12 col-md-6 offset-md-3">
-        Withdraw successful! View Transaction:
+        {{ $t('pages.staking.withdraw_successful') }}:
         <transaction-field :transaction-hash="resultHash" />
     </div>
 </div>
```
Comment (karynemayer) [inline src/pages/staking/WithdrawPage.vue:77]: `console.error` message

**Review by karynemayer — CHANGES_REQUESTED** [review]
Please review the lines listed (I couldn't put a comment in the lines that weren't active on the diff :( ). I also mentioned `console.error` messages that might need to be showed in the UI and thus, translated.

File: package.json [hunk 26]
```diff
@@ -41,11 +41,12 @@
     "ual-wombat": "^0.3.3",
     "universal-authenticator-library": "^0.3.0",
     "vue": "3",
+    "vue-i18n": "^9.0.0",
     "vue-json-viewer": "3",
     "vue-router": "4",
     "vue3-click-away": "^1.2.4",
     "vuex": "4",
-    "web3": "^1.8.1"
+    "web3": "^1.8.1"
```
[the remark this answers is not in the document]
Comment (Viterbo) [reply package.json:49]: fixed

File: src/components/ConfirmationDialog.vue [hunk 27]
```diff
@@ -62,5 +62,5 @@ q-dialog( v-model="showDialog" persistent)
           span.q-ml-sm {{ dialogMessage }}
         q-card-actions(align="right")
           q-btn(flat label="Dismiss" color="primary" v-close-popup)
-          q-btn(v-if="!status" flat label="Verify Contract" color="primary" v-close-popup @click="navigate")
+          q-btn(v-if="!status" flat :label="$t('components.verify_contract')" color="primary" v-close-popup @click="navigate")
```
[the remark this answers is not in the document]
Comment (Viterbo) [reply src/components/ConfirmationDialog.vue:65]: fixed

File: src/components/HomeInfo.vue [hunk 28]
```diff
@@ -69,7 +69,7 @@ export default {
         .col-8.q-pl-sm
             .col-12
               .column.text-subtitle2
-                | Latest Block
```
[the remark this answers is not in the document]
Comment (Viterbo) [reply src/components/HomeInfo.vue:72]: fixed

File: src/components/Transaction/LogsViewer.vue (hunk 1 above)
Comment (Viterbo) [reply src/components/Transaction/LogsViewer.vue:25]: fixed

File: src/components/TransactionTable.vue (hunk 2 above)
Comment (Viterbo) [reply src/components/TransactionTable.vue:74]: I found this that I forgot to transtale
```js
    const tableTitle = computed(() =>
      isTransaction.value ? 'Actions' : 'Latest Transactions'
    );
```

But I couldn't find the three changes that you pointed out. There's only code with no text surrounding those lines. Maybe you refer to another file. Can you check that?

File: src/components/TransferTable.vue (hunk 3 above)
Comment (Viterbo) [reply src/components/TransferTable.vue:76]: fixed

File: src/components/TransferTable.vue (hunk 4 above)
Comment (Viterbo) [reply src/components/TransferTable.vue:163]: fixed

File: src/components/Health/Monitor.vue [hunk 29]
```diff
@@ -35,6 +13,29 @@ export default {
...
+            {
+                name: 'message',
+                label: this.$t('components.health.message'),
+                align: 'left',
+            },
+        ]
+
         return {
             rows: [],
             tasks: [],
```
[the remark this answers is not in the document]
Comment (Viterbo) [reply src/components/Health/Monitor.vue:41]: fixed

File: src/components/inputs/AddressInput.vue [hunk 30]
```diff
@@ -42,9 +42,9 @@ export default {
     },
```
[the remark this answers is not in the document]
Comment (Viterbo) [reply src/components/inputs/AddressInput.vue:42]: fixed

File: src/components/inputs/AddressInput.vue (hunk 5 above)
Comment (Viterbo) [reply src/components/inputs/AddressInput.vue:50]: fixed

File: src/pages/AccountAddress.vue (hunk 6 above)
Comment (Viterbo) [reply src/pages/AccountAddress.vue:134]: fixed

File: src/pages/Block.vue (hunk 7 above)
Comment (Viterbo) [reply src/pages/Block.vue:48]: fixed

File: src/pages/ContractVerification.vue (hunk 8 above)
Comment (Viterbo) [reply src/pages/ContractVerification.vue:161]: fixed

File: src/pages/Transaction.vue (hunk 9 above)
Comment (Viterbo) [reply src/pages/Transaction.vue:131]: fixed

File: src/pages/Transaction.vue (hunk 10 above)
Comment (Viterbo) [reply src/pages/Transaction.vue:146]: fixed

File: src/pages/Transaction.vue (hunk 11 above)
Comment (Viterbo) [reply src/pages/Transaction.vue:241]: fixed

File: src/pages/Transaction.vue (hunk 12 above)
Comment (Viterbo) [reply src/pages/Transaction.vue:328]: fixed

File: src/pages/Transaction.vue (hunk 13 above)
Comment (Viterbo) [reply src/pages/Transaction.vue:329]: I fixed all the texts that looks like `{{ `From: ` }}`

File: src/pages/Transaction.vue (hunk 14 above)
Comment (Viterbo) [reply src/pages/Transaction.vue:364]: fixed

File: src/pages/Transaction.vue (hunk 15 above)
Comment (Viterbo) [reply src/pages/Transaction.vue:372]: fixed

File: src/pages/staking/StakeForm.vue (hunk 17 above)
Comment (Viterbo) [reply src/pages/staking/StakeForm.vue:57]: fixed

File: src/pages/staking/StakeForm.vue (hunk 16 above)
Comment (Viterbo) [reply src/pages/staking/StakeForm.vue:13]: fixed

File: src/pages/staking/StakeForm.vue (hunk 18 above)
Comment (Viterbo) [reply src/pages/staking/StakeForm.vue:259]: fixed

File: src/pages/staking/Staking.vue (hunk 19 above)
Comment (Viterbo) [reply src/pages/staking/Staking.vue:56]: Wow!! there were so many XD

fixed

File: src/pages/staking/StakingStats.vue (hunk 20 above)
Comment (Viterbo) [reply src/pages/staking/StakingStats.vue:150]: fixed

File: src/pages/staking/UnstakeForm.vue (hunk 22 above)
Comment (Viterbo) [reply src/pages/staking/UnstakeForm.vue:46]: fixed

File: src/pages/staking/UnstakeForm.vue (hunk 21 above)
Comment (Viterbo) [reply src/pages/staking/UnstakeForm.vue:32]: fixed

File: src/pages/staking/UnstakeForm.vue (hunk 23 above)
Comment (Viterbo) [reply src/pages/staking/UnstakeForm.vue:180]: fixed

File: src/pages/staking/UnstakeForm.vue (hunk 24 above)
Comment (Viterbo) [reply src/pages/staking/UnstakeForm.vue:205]: fixed

File: src/pages/staking/WithdrawPage.vue (hunk 25 above)
Comment (Viterbo) [reply src/pages/staking/WithdrawPage.vue:77]: fixed

File: src/boot/i18n.js [hunk 31]
```diff
@@ -0,0 +1,16 @@
...
+        // locale: 'en-us',
+        locale: 'es-es',
+        globalInjection: true,
+        messages,
+    });
+
+    // Set i18n instance on app
+    app.use(i18n);
+    // app.config.globalProperties.$t = i18n?;
+});
```
[the remark this answers is not in the document]
Comment (Viterbo) [reply src/boot/i18n.js:14]: I create an issue for that: https://github.com/telosnetwork/teloscan/issues/304

**Review by donnyquixotic — DISMISSED** [review]

File: src/components/inputs/BooleanArrayInput.vue [hunk 32]
```diff
@@ -74,7 +74,8 @@ export default {
         },
         shapedLabel() {
             const size = (Number.isInteger(+this.size) && +this.size !== -1) ? `${+this.size}` : '';
-            return `${this.label} (bool[${size}])`
+            // return `${this.label} (bool[${size}])`
```
Comment (EJSG) [inline src/components/inputs/BooleanArrayInput.vue:77]: rm comment

File: src/pages/AccountAddress.vue [hunk 33]
```diff
@@ -142,12 +142,13 @@ export default {
                     decimalIndex + 5,
                 )}`;
             }
-            return `${strBalance} TLOS`;
+            // return `${strBalance} TLOS`;
```
Comment (EJSG) [inline src/pages/AccountAddress.vue:145]: rm comment

File: src/pages/AccountAddress.vue [hunk 34]
```diff
@@ -142,12 +142,13 @@ export default {
...
+            return this.$('pages.tlos_balance', { balance: strBalance });
         },
         getAddressNativeExplorerURL() {
             if (!this.telosAccount) return '';
 
-            return `${process.env.NETWORK_EXPLORER}/account/${this.telosAccount}`;
+            return this.$('pages.account_url', { domain: process.env.NETWORK_EXPLORER, account: this.telosAccount });
```
Comment (EJSG) [inline src/pages/AccountAddress.vue:151]: should we be using i18n for URLs? does this mean the expectation is that all URLs will now be localized? I don't think that's necessary

File: src/pages/staking/StakeForm.vue [hunk 35]
```diff
@@ -203,28 +207,26 @@ export default {
...
+                    return this.$t('pages.staking.insufficient_tlos_balance');
                 else if(this.isNative)
-                    return 'Login using an EVM wallet'
+                    return this.$t('pages.staking.login_using_an_evm_wallet');
                 else
                     return '';
             }
 
-            return 'Wallet not connected';
+            // return 'Wallet not connected';
```
Comment (EJSG) [inline src/pages/staking/StakeForm.vue:224]: rm comment

File: src/pages/staking/StakeForm.vue [hunk 36]
```diff
@@ -155,17 +155,21 @@ export default {
         MetaMaskLogo,
         displayConfirmModal: false,
         resultHash: null,
-        header: 'Stake TLOS',
-        subheader: 'Staking your TLOS to sTLOS grants you access to a steady income and various DeFi applications, ' +
+        harcoded_header: 'Stake TLOS',
```
Comment (EJSG) [inline src/pages/staking/StakeForm.vue:158]: why are some of these left as "hardcoded"?

**Review by EJSG — CHANGES_REQUESTED** [review]
this is a lot of work, nicely done. I also appreciate that you added notifications for error messages :)
i have some initial feedback, and also I am having issues with the branch in general (we will talk about that over telegram)

File: src/pages/staking/StakeForm.vue (hunk 36 above)
Comment (Viterbo) [reply src/pages/staking/StakeForm.vue]: It was a helper I used to "teach" my Github copilot plugin which uses AI to suggest code ahead.

Changing all the `label`(s) for `harcoded_label`(s) and immediately after the first one, writing the correct translation solution, my copilot learned from it and write the rest for me.

But I just forgot to remove them.

Done ([735c2d2](https://github.com/telosnetwork/teloscan/pull/292/commits/735c2d2f6ddcd35dbada25260a3b4c48425873e8))

File: src/components/inputs/BooleanArrayInput.vue (hunk 32 above)
Comment (Viterbo) [reply src/components/inputs/BooleanArrayInput.vue:77]: fixed

File: src/pages/AccountAddress.vue (hunk 33 above)
Comment (Viterbo) [reply src/pages/AccountAddress.vue:145]: fixed

File: src/pages/AccountAddress.vue (hunk 34 above)
Comment (Viterbo) [reply src/pages/AccountAddress.vue]: Yes, @karynemayer suggested we should leave no string behind. Even if it does not need translation like urls or balance symbols (see `tlos_balance: '{ balance } TLOS'`)

File: src/pages/staking/StakeForm.vue (hunk 35 above)
Comment (Viterbo) [reply src/pages/staking/StakeForm.vue:217]: fixed

**Review by donnyquixotic — APPROVED** [review]

**Comment by EJSG** [comment]
The address page should say "Account" at the top for accounts, 
image
compare to https://www.teloscan.io/address/0x160505F3dfD1cb58B91e322c828Ae0F74c043c3C

**Comment by EJSG** [comment]
contract tab is missing some localizations on route `/address/0xf42C4b1D1F93e9e31208Ae6907546fF0F675205b#contract`:
image

**Comment by EJSG** [comment]
tx details page, route `/tx/0x59feb330964cc8d3922cbb22207dc0070b1d744bf865ba6726dba79d331cfd29#general` 

image

as well as Details tab
image

**Review by EJSG — APPROVED** [review]
very nice work

**Review by karynemayer — APPROVED** [review]

**Pull request merged**